src/Hazelcast.Net.Tests/Serialization/ServerDeserializationTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Hazelcast.Testing;
+using Hazelcast.Testing.Remote;
+using NUnit.Framework;
+
+namespace Hazelcast.Tests.Serialization
+{
+    /// <summary>
+    /// Checks that object is deserialized server-side (Java) to the same value as was serialized client-side (.NET).
+    /// </summary>
+    [TestFixture]
+    public class ServerDeserializationTests : SingleMemberClientRemoteTestBase
+    {
+        [Test]
+        public async Task GuidTest()
+        {
+            var expected = Guid.NewGuid();
+            var actualStr = await GetAsServerString(expected);
+
+            Assert.AreEqual(expected, Guid.Parse(actualStr));
+        }
+
+        private async Task<string> GetAsServerString<T>(T value, [CallerMemberName] string callerName = default)
```

Comment:
Can't we just use `CreateUniqueName()` to get a unique map name and avoid the `[CallerMemberName]` thing? I think it adds unneeded complexity.

Reply:
The parameter is still needed to pass map name, just thought that using `[CallerMemberName]` would be simpler.

Reply:
Nevermind, it's not needed.